lib/mail/renderers/rfc_2822.ex

```diff
@@ -152,6 +153,14 @@ defmodule Mail.Renderers.RFC2822 do
   defp render_subtypes([{key, value} | subtypes]) do
     key = String.replace(key, "_", "-")
     value = encode_header_value(value, :quoted_printable)
+
+    value =
+      if value =~ ~r/[\s;]/ do
```

Comment:
We should match all the tspecial characters of RFC2045
`if Regex.match?(~r/[\s()<>@,;:\\<\/\[\]?=]/, value) do`